test/e2e/operator_test.go

```diff
@@ -228,3 +252,195 @@ func setImage(deployment *appsv1.Deployment, image string) {
 		}
 	}
 }
+
+func TestDNSForwarding(t *testing.T) {
+	cl, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create the upstream resolver ConfigMap.
+	upstreamPodName := "test-upstream"
+	upstreamPodNS := "openshift-dns"
+	upstreamCfgMap := upstreamConfigMap(upstreamPodName, upstreamPodNS)
+	if err := cl.Create(context.TODO(), upstreamCfgMap); err != nil {
+		t.Fatalf("failed to create configmap %s/%s: %v", upstreamCfgMap.Namespace, upstreamCfgMap.Name, err)
+	}
+	defer func() {
+		if err := cl.Delete(context.TODO(), upstreamCfgMap); err != nil {
+			t.Fatalf("failed to delete configmap %s/%s: %v", upstreamCfgMap.Namespace, upstreamCfgMap.Name, err)
+		}
+	}()
+
+	// Get the CoreDNS image used by the test upstream resolver.
+	co := &configv1.ClusterOperator{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: operatorcontroller.DNSOperatorName}, co); err != nil {
+		t.Fatalf("failed to get cluster operator dns: %v", err)
+	}
+	var (
+		upstreamImage  string
+		coreImageFound bool
+	)
+	if co.Status.Versions != nil {
+		for _, ver := range co.Status.Versions {
```

Comment:
You can omit the nil check; `for` will iterate 0 times if the slice is nil.

Reply:
ack